### PR TITLE
Fix/refactor footer button

### PR DESCRIPTION
**Pull request by Oliversw**
- Refactored the footer so that the buttons are now links that work with react router
- Swapped placement of 'back' and 'next' buttons since client in the UK reads from left to right
- Added statements to make the 'review scenario' and 'next' buttons only load if the props are supplied (for methods page and select your play page)
- Changed footer to a functional component since it does not need state
- Added prop-type checking
- Changed the declaration of footer on existing pages to reflect the above changes and linked them together. Still missing some pieces though

#50

### CODE REVIEW

**Review by hshahwan — APPROVED** [review]

File: client/src/Component/Common/Footer/index.js [hunk 1]
```diff
@@ -1,27 +1,38 @@
...
   PreviousButton,
-  BackToScenerio,
+  BackToScenario,
   Line,
 } from './Footer.style';
 
-class Footer extends Component {
-  state = {};
+const Footer = ({ nextLink, backLink, reviewScenario }) => {
+  // const { nextLink, backLink, reviewScenario } = props;
```
Comment (ali-7) [inline client/src/Component/Common/Footer/index.js]: remove this line

File: client/src/Component/Common/Footer/index.js [hunk 2]
```diff
@@ -1,27 +1,38 @@
...
-    );
-  }
-}
+  return (
+    <React.Fragment>
+      <Line />
+      <Buttons>
+        <PreviousButton as={Link} to={backLink} type="button" />
+        {reviewScenario && <BackToScenario>Review Scenario</BackToScenario>}
+        { nextLink && <NextButton as={Link} to={nextLink} type="button" />}
```
Comment (ali-7) [inline client/src/Component/Common/Footer/index.js]: `{ nextLink` >>> `{nextLink` :)))))

File: client/src/Component/views/Methods/index.js [hunk 3]
```diff
@@ -7,7 +7,7 @@ const Methods = () => {
     <div>
       <Header headerImg={null} titleText="Select your methods" />
       <h1>Hello from Methods.js</h1>
-      <Footer />
+      <Footer backLink="/play" nextLink="/" reviewScenario={true} />
```
Comment (ali-7) [inline client/src/Component/views/Methods/index.js]: `reviewScenario={true}`  when passed as `reviewScenario` only, it will be true

**Review by ali-7 — CHANGES_REQUESTED** [review]

File: client/src/Component/views/Play/index.js [hunk 4]
```diff
@@ -7,7 +7,7 @@ const Play = () => {
     <div>
       <Header headerImg={null} titleText="Choose your game play" />
       <h1>Choose your game play:</h1>
-      <Footer />
+      <Footer backLink="/howto" nextLink={false} reviewScenario={false} />
```
Comment (ali-7) [inline client/src/Component/views/Play/index.js]: `nextLink={false}`  >>> `nextLink={null}` or `nextLink=""` to be valid props.

**Review by ali-7 — CHANGES_REQUESTED** [review]

**Review by teenie-quaggard — APPROVED** [review]

**Review by hshahwan — APPROVED** [review]

**Review by ali-7 — APPROVED** [review]
:+1:

**Pull request merged**